python/eups/stack/ProductStack.py

```diff
@@ -252,8 +252,15 @@ def save(self, flavors=None, dir=None):
             raise CacheOutOfSync(outofsync)
 
     def _cacheFileIsInSync(self, file):
-        return (file not in self.modtimes or
-                os.stat(file).st_mtime <= self.modtimes[file])
+        if file not in self.modtimes:
+            return True
+        try:
+            older = os.stat(file).st_mtime <= self.modtimes[file]
```

Comment:
There seem to be a number of similar `os.stat()` calls.  They don't need protection as well?

It's curious that the only thing that seems to remove this file is an admin command.

Reply:
I will take a look at the other places.

I was also wondering how the file disappears even in the case where 30 EUPS processes in parallel are trying to read the same database file. The atomic rename should not result in the file disappearing so I realize that this is papering over the cracks a little. @RobertLuptonTheGood  do you know of a way that the pickle file could be deleted in normal usage?

Reply:
I've added a couple more traps but `reload` is a problem.